independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/handlers/ClientSendRequestHandler.java

```diff
@@ -70,7 +73,30 @@ public void handle(RestClientRequestContext requestContext) {
             return;
         }
         requestContext.suspend();
-        Uni<HttpClientRequest> future = createRequest(requestContext);
+        Uni<HttpClientRequest> future = createRequest(requestContext)
+                .runSubscriptionOn(new Executor() {
+                    @Override
+                    public void execute(Runnable command) {
+                        Context current = Vertx.currentContext();
+                        ClientRequestContextImpl clientRequestContext = requestContext.getClientRequestContext();
```

Comment:
can `getClinetRequestContext` ever return null here?

Reply:
I believe not. The `getOrCreateClientRequestContext` is called before the client request execution.

Reply:
there is a test where it got `null` :-D